editor/editor_properties.cpp

```diff
@@ -1796,12 +1796,21 @@ EditorPropertyColor::EditorPropertyColor() {
 void EditorPropertyNodePath::_node_selected(const NodePath &p_path) {
 
 	NodePath path = p_path;
-	Node *base_node = Object::cast_to<Node>(get_edited_object());
-	if (base_node == NULL && get_edited_object()->has_method("get_root_path")) {
-		base_node = get_edited_object()->call("get_root_path");
+	Object *edited_object = get_edited_object();
+	if (edited_object->get_class() == "EditorPropertyArrayObject") {
+		edited_object = parent_object;
 	}
+
+	Node *base_node = Object::cast_to<Node>(edited_object);
+	if (base_node && edited_object && edited_object->has_method("get_root_path")) {
```

Comment:
The previous code was `base_node == NULL`, so this seems like a typo? If not, the original cast is not useful as it's being overridden.